### PR TITLE
Fixes #9997: Document the relay API

### PR DESCRIPTION
**Pull request by amousset**
https://www.rudder-project.org/redmine/issues/9997

### CODE REVIEW

**Comment by amousset** [comment]
Commit modified

File: 8_reference/80_relay_api.txt [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+===== Description
+
+The remote run API applies to all nodes that are below the target relay server, which means:
+
+* All nodes directly connected to the server
+* All the relays that have the target node as policy server, and all nodes that are below them
+
+In particular, it does not act on the target relay itself (except for the root server which is its own policy server).
+
+There are different methods, whether you want to trigger all nodes, and only a part of them.
```
Comment (peckpeck) [inline 8_reference/80_relay_api.txt]: s/and/or/

File: 8_reference/80_relay_api.txt [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+===== Security
+
+The remote run calls are not authenticated, but restricted to:
+
+* Local calls
+* The target's policy server
```
Comment (peckpeck) [inline 8_reference/80_relay_api.txt]: the target is ambiguous, prefer the term node for the remote run target, and relay for the api target

File: 8_reference/80_relay_api.txt [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+The relay that receives an API call to share a file will:
+
+* share the file directly if the target nodes is one of its managed nodes.
+* send the file to a sub-relay if the target is somewhere under it
+* forward the file to its policy server if the target is nowhere under it
+
+The relay that receive a *HEAD* call will:
+
+* If the file exists, compare the provided hash with the hash of the stored file, and return the result
+* If the file
```
Comment (peckpeck) [inline 8_reference/80_relay_api.txt]: missing sentence

**Comment by amousset** [comment]
Commit modified

**Pull request merged**